scripts/fix/diplomats.lua

```diff
@@ -0,0 +1,103 @@
+-- Add Elven diplomats to negotiate tree caps
+--[[=begin
+
+fix/diplomats
+=============
+Adds a Diplomat position to all Elven civilizations, allowing them to negotiate
+tree cutting quotas - and you to violate them and start wars.
+This was vanilla behaviour until ``0.31.12``, in which the "bug" was "fixed".
+
+=end]]
+
+
+function update_pos(pos, ent) do
+    -- update the given position on the given entity
+    if not pos then
+        -- there was no diplomat - create it
+        pos = df.entity_position:new()
+        ent.positions.own:insert(0, pos)
+
+        pos.code = "DIPLOMAT"
+        pos.id = ent.positions.next_position_id + 1
+        pos.flags.set(entity_position_flags.DO_NOT_CULL)
+        pos.flags.set(entity_position_flags.MENIAL_WORK_EXEMPTION)
+        pos.flags.set(entity_position_flags.SLEEP_PRETENSION)
+        pos.flags.set(entity_position_flags.PUNISHMENT_EXEMPTION)
+        pos.flags.set(entity_position_flags.ACCOUNT_EXEMPT)
+        pos.flags.set(entity_position_flags.DUTY_BOUND)
+        pos.flags.set(entity_position_flags.COLOR)
+        pos.flags.set(entity_position_flags.HAS_RESPONSIBILITIES)
+        pos.flags.set(entity_position_flags.IS_DIPLOMAT)
+        pos.flags.set(entity_position_flags.IS_LEADER)
+        -- not sure what these flags do, but the game sets them for a valid diplomat
+        pos.flags.set(entity_position_flags.unk_12)
+        pos.flags.set(entity_position_flags.unk_1a)
+        pos.flags.set(entity_position_flags.unk_1b)
+        pos.name[0] = "Diplomat"
```

Comment:
...looks like I didn't push the latest version of `fix/diplomats`.  This is fixed in the merchants script, though.  Give me a couple of hours and I'll do some last fixes for both.